llvm/docs/TestSuiteGuide.md

```diff
@@ -352,6 +352,27 @@ performance. You can find more information at
 of the LLVM project is hosted at [http://lnt.llvm.org](http://lnt.llvm.org).
 
 
+Included Test Suites
+--------------------
+
+The LLVM Test Suite repository includes a number of test suites that we have
+copied into the repository from external sources, and hand-written CMake files
+to hook into the LLVM Test Suite system. For example, these include GCC's C
+and Fortran Torture test suites.
+
+In some cases, these external tests have a different license to other code in
+the llvm-test-suite repository, which must be documented in `LICENSE.TXT` in the
+root of the test suite repository, under the heading "Software from third
+parties included in the LLVM Project".
+
+The only changes we can accept to the tests in these suites are where we bring
+in all the changes from the external sources. Users should not make individual
+downstream changes to these tests.
```

Comment:
I think "users" and "downstream" might be a bit ambiguous in this context. Unless I'm getting the wrong end of the stick, I think you're referring to LLVM contributors making changes and attempting to submit them to the llvm-test-suite repo (downstream from the original tests/benchmarks, but "upstream" from many people's point of view!). I assume we're not trying to pass judgement on whatever atrocities people apply to an llvm-test-suite checkout on their own machine for their downstream only purposes(!).

I have a more general comment about the policy I'll leave on the Discourse thread.

Reply:
You're right that this is ambiguous. I think the consensus in Discourse is that this needs significant rewriting anyway, but I will make sure I am clearer about what I'm referring to.